### PR TITLE
Scalar mul assign

### PR DESCRIPTION
**Pull request by 3pmile**
**Description**

This PR implements MulAssign

for scalars of 

- [ ] MatZ
- [ ] PolyOverZ
- [ ] MatPolyOverZ

- [ ] PolyOverZq
- [ ] MatZq

- [ ] PolyOverQ
- [ ] MatQ

**Testing**

- [ ] I added basic working examples (possibly in doc-comment)
- [ ] I added tests for large (pointer representation) values
- [ ] I triggered all possible errors in my test in every possible way
- [ ] I included tests for all reasonable edge cases

**Checklist:**

- [ ] I have performed a self-review of my own code
  - [ ] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [ ] The chosen implementation is not more complex than it has to be
  - [ ] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [ ] The doc comments fit our style guide
  - [ ] I have credited related sources if needed

### CODE REVIEW

File: src/rational/poly_over_q/arithmetic/div_scalar.rs [hunk 1]
```diff
@@ -83,12 +96,71 @@ impl Div<&Q> for &PolyOverQ {
...
+    /// # Examples
+    /// ```
+    /// use qfall_math::rational::{Q, PolyOverQ};
+    /// use std::str::FromStr;
+    ///
+    /// let mut polyq = PolyOverQ::from_str(&format!("3  1/3 -5 99/2")).unwrap();
+    /// let q = Q::from((3, 4));
+    ///
+    /// polyq /= &q;
+    /// ```
```
[the remark this answers is not in the document]
Comment (Marvin-Beckmann) [reply src/rational/poly_over_q/arithmetic/div_scalar.rs]: I do not consider this actual scalar multiplication, but I added more examples
